client/my-sites/theme/live-preview-button/index.tsx

```diff
@@ -19,17 +19,24 @@ export const LivePreviewButton: FC< Props > = ( { themeId, siteId } ) => {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
 
+	const [ isLoading, setIsLoading ] = useState( false );
+
 	const isLivePreviewSupported = useSelector( ( state ) =>
 		getIsLivePreviewSupported( state, themeId, siteId )
 	);
 	if ( ! isLivePreviewSupported ) {
 		return null;
 	}
 
+	const handleClick = () => {
+		setIsLoading( true );
+		dispatch( livePreview( themeId, siteId, 'detail' ) );
+	};
```

Comment:
Just to hear your opinion, but should we implement error handling and terminate the loading state at this point? For example, in cases where the theme installation fails.

Reply:
I try to time-box this but it is complicated as there is currently nothing in the codebase that listens to the `THEME_INSTALL_FAILURE` action 😭 

We can follow-up in a future PR, what do you think?